### PR TITLE
[XM] More xtro-sharpie fixes

### PR DESCRIPTION
**Pull request by chamons**

### CODE REVIEW

File: src/CoreBluetooth/Enums.cs [hunk 1]
```diff
@@ -23,6 +24,17 @@ public enum CBManagerState : nint {
 		PoweredOff,
 		PoweredOn
 	}
+#else
```
Comment (chamons) [inline src/CoreBluetooth/Enums.cs]: Do not love this solution at all. Open to better ones.

File: src/avfoundation.cs [hunk 2]
```diff
@@ -1823,7 +1823,7 @@ interface AVAudioSession {
 		[Notification (typeof (AVAudioSessionSecondaryAudioHintEventArgs))]
 		NSString SilenceSecondaryAudioHintNotification { get; }
 		
-		[NoWatch, NoTV, iOS(10,0), Mac (10,12)]
+		[NoWatch, NoTV, iOS(10,0)]
```
Comment (spouliot) [inline src/avfoundation.cs]: diff lack context, should it be a `[NoMac]` ?

File: src/AVFoundation/Enums.cs [hunk 3]
```diff
@@ -718,11 +718,13 @@ public enum AVPlayerTimeControlStatus : nint
 		Playing
 	}
 
-	[NoWatch, NoTV, iOS (10,0), Mac (10,12)]
+#if !MONOMAC || !XAMCORE_4_0
```
Comment (spouliot) [inline src/AVFoundation/Enums.cs]: not sure I get the condition (wrt 4_0) ? could not it just be `[NoMac]` ?

File: src/CoreBluetooth/Enums.cs (hunk 1 above)
Comment (spouliot) [inline src/CoreBluetooth/Enums.cs]: are the values are needed at build time ?

Might be better to have if `#if` only on the `enum` line - that way there's not chance of only one side of the (identical) values being updated in the future

File: src/corebluetooth.cs [hunk 4]
```diff
@@ -34,13 +34,15 @@ interface CBCentralInitOptions {
 #endif
 	}
 
+#if !MONOMAC
```
Comment (spouliot) [inline src/corebluetooth.cs]: weird, the `[NoMac]` should have solve that

File: tests/xtro-sharpie/osx.pending [hunk 5]
```diff
@@ -196,4 +196,4 @@
 !unknown-native-enum! NSWindowNumberListOptions bound
 !unknown-native-enum! NSType bound
 !unknown-native-enum! SLComposeViewControllerResult bound
-
+!wrong-base-type! MDLVoxelArray expected MDLObject actual NSObject
```
Comment (spouliot) [inline tests/xtro-sharpie/osx.pending]: comments are always helpful ;-)

File: tests/xtro-sharpie/osx.ignore [hunk 6]
```diff
@@ -388,3 +388,7 @@
 
 # 32-bit only, and marked correctly in binding
 !unknown-type! NSMenuView bound
+
+# Not found anywhere in xamarin-macios / maccore
+!unknown-pinvoke! SSLSetSessionStrengthPolicy bound
```
Comment (spouliot) [inline tests/xtro-sharpie/osx.ignore]: `src/Security/SslContext.cs` (at least on master) has this defined

File: src/CoreBluetooth/Enums.cs (hunk 1 above)
Comment (chamons) [inline src/CoreBluetooth/Enums.cs]: Yes. The next two enums "steal" the values.

Good idea on sharing the guts of the enum.

File: src/AVFoundation/Enums.cs (hunk 3 above)
Comment (chamons) [inline src/AVFoundation/Enums.cs]: APIDiff claimed it was an API removal when i tested it. But as you point out, that doesn't make any sense. Let me validate against 2.10 and fix.

File: tests/xtro-sharpie/osx.pending (hunk 5 above)
Comment (chamons) [inline tests/xtro-sharpie/osx.pending]: Part of the "# Fixed in XAMCORE_4_0" block outside of diff.

File: src/avfoundation.cs (hunk 2 above)
Comment (chamons) [inline src/avfoundation.cs]: AVAudioSession is in a giant #if !MONOMAC block already.

**Comment by spouliot** [comment]
LGTM if bots are happy :)

**Comment by chamons** [comment]
Real build failure, investigating:

DesignatedInitializer crashed.

**Pull request merged**